app/(app)/offence/result.tsx

```diff
@@ -0,0 +1,118 @@
+import { router, useLocalSearchParams, useNavigation } from 'expo-router'
+import { useEffect } from 'react'
+import { useTranslation } from 'react-i18next'
+import { View } from 'react-native'
+
+import ContinueButton from '@/components/navigation/ContinueButton'
+import ContentWithAvatar from '@/components/screen-layout/ContentWithAvatar'
+import ErrorScreen from '@/components/screen-layout/ErrorScreen'
+import ScreenViewCentered from '@/components/screen-layout/ScreenViewCentered'
+import IconButton from '@/components/shared/IconButton'
+import { useClearHistory } from '@/hooks/useClearHistory'
+import { getRoleByKey } from '@/modules/backend/constants/roles'
+import { ScanResultEnum } from '@/modules/backend/openapi-generated'
+import { getDefaultOffenceStateByRole } from '@/state/OffenceStore/getDefaultOffenceStateByRole'
+import { useOffenceStoreContext } from '@/state/OffenceStore/useOffenceStoreContext'
+import { useSetOffenceState } from '@/state/OffenceStore/useSetOffenceState'
+
+type OffenceResultSearchParams = {
+  scanResult: ScanResultEnum
+}
+
+const OffenceResultPage = () => {
+  const { t } = useTranslation()
+  const { scanResult } = useLocalSearchParams<OffenceResultSearchParams>()
+
+  const { resetOffenceState } = useSetOffenceState()
+  const { roleKey, zonePhoto, zone, location } = useOffenceStoreContext((state) => state)
+  const role = getRoleByKey(roleKey)
+
+  const clearHistory = useClearHistory()
+  const navigation = useNavigation()
+
+  const headerRight = () => (
+    <IconButton
+      name="home"
+      accessibilityLabel={t('offenceResult.home')}
+      onPress={() => {
+        router.replace('/')
+        clearHistory()
+      }}
+    />
+  )
+
+  // Function to reset the navigation stack to the 'scan/licence-plate-camera' screen when going back (back button or swipe back)
+  useEffect(() => {
+    if (!(scanResult && role)) return () => {}
+
+    return navigation.addListener('beforeRemove', (e) => {
+      // prevents going back
+      e.preventDefault()
+
+      const state = navigation.getState()
+
+      const index = state.routes.findIndex((route) => route.name === 'scan/licence-plate-camera')
+      const currentRoute = state.routes[state.index]
+
+      // resets navigation to remove all the routes between the 'scan/licence-plate-camera' screen and the current screen
+      navigation.reset({
```

Comment:
According to the docs, this might not be necessary It says here ,that[ if you navigate to a route that is already in the stack, it would unwind and go back to that route. ](https://docs.expo.dev/router/navigating-pages/#understanding-native-navigation) Would you be able to verify this behaviour and if it's the case use that?  🙇